runtime/sys.c

```diff
@@ -601,7 +601,7 @@ CAMLprim value caml_sys_time(value unit)
 #ifdef _WIN32
 extern int caml_win32_random_seed (intnat data[16]);
 #else
-int caml_unix_random_seed(intnat data[16])
+int caml_unix_random_seed(intnat data[static 16])
```

Comment:
This particular change seem to be orthogonal to that PR. Is it intentional, and if so, what is the rationale behind it and why isn't there a similar change in the win32 code path?

Reply:
if only… [C11: static inside array parameter square brackets](https://developercommunity.visualstudio.com/t/C11:-static-inside-array-parameter-squar/1475168)

MSVC doesn't support this C99 feature that tells that the array being passed contains at least 16 elements, which helps with type/bounds checking. I had added it to both code paths, but had to remove it on win32.

Reply:
Can you name any C compiler/linter/checker which actually makes use of the `static` keyword in array sizes and does not only parse it? This is yet another C99 feature which looks useful on paper but hasn't found its use.

Reply:
Yes. Both gcc and clang warn if `NULL` or if an array too small is passed to the function. Interestingly gcc also warns in the non-static case if an array too small is passed. cppcheck considers in both cases that the array has a fixed size and warn of out-of-bounds accesses.

<details>

```c
void f(int a[10]);
void g(int a[static 10]);
void h(int a[static 10]) {
    a[42] = 42;
}

void m(void) {
    int a[8];
    int b[11];

    f(0);
    f(a);
    f(b);

    g(0);
    g(a);
    g(b);
}
```

```console
$ /opt/homebrew/Cellar/gcc/14.2.0_1/bin/gcc-14  -O0 -Wall -Wextra -D_FORTIFY_SOURCE=3 -c test.c
test.c: In function 'm':
test.c:15:5: warning: argument 1 null where non-null expected [-Wnonnull]
   15 |     g(0);
      |     ^
test.c:2:6: note: in a call to function 'g' declared 'nonnull'
    2 | void g(int a[static 10]);
      |      ^
test.c:12:5: warning: 'f' accessing 40 bytes in a region of size 32 [-Wstringop-overflow=]
   12 |     f(a);
      |     ^~~~
test.c:12:5: note: referencing argument 1 of type 'int[10]'
test.c:1:6: note: in a call to function 'f'
    1 | void f(int a[10]);
      |      ^
test.c:16:5: warning: 'g' accessing 40 bytes in a region of size 32 [-Wstringop-overflow=]
   16 |     g(a);
      |     ^~~~
test.c:16:5: note: referencing argument 1 of type 'int[10]'
test.c:2:6: note: in a call to function 'g'
    2 | void g(int a[static 10]);
      |      ^
$ /opt/homebrew/Cellar/llvm/19.1.2/bin/clang -O0 -Wall -Wextra -D_FORTIFY_SOURCE=3 -c test.c
test.c:15:5: warning: null passed to a callee that requires a non-null argument [-Wnonnull]
   15 |     g(0);
      |     ^ ~
test.c:2:12: note: callee declares array parameter as static here
    2 | void g(int a[static 10]);
      |            ^~~~~~~~~~~~
test.c:16:5: warning: array argument is too small; contains 8 elements, callee requires at least 10 [-Warray-bounds]
   16 |     g(a);
      |     ^ ~
test.c:2:12: note: callee declares array parameter as static here
    2 | void g(int a[static 10]);
      |            ^~~~~~~~~~~~
2 warnings generated.
```

</details>